File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function (nums) {
    let j = 1;
    for (let i = 0; i < nums.length; i++) {
        if (nums[j - 1] != nums[i]) {
            nums[j] = nums[i];
            j++;
        }
    }
    return j;
};

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int left = 0;
        int right = 1;

        while (left<nums.length) {
            if (nums[right-1] != nums[left]) {
                nums[right++]=nums[left++];
            } else {
                left++;
            }
        }
        return right;
    }
}